File: docs/components_page/components/modal/fullscreen.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

modal <- htmlDiv(
  list(
    dbcButton("Open modal", id = "open-fs"),
    dbcModal(
      list(
        dbcModalHeader(dbcModalTitle("Fullscreen modal")),
        dbcModalBody("Wow this thing takes up a lot of space...")
      ),
      id = "modal-fs",
      fullscreen = TRUE
    )
  )
)


app$callback(
  output("modal-fs", "is_open"),
  list(
    input("open-fs", "n_clicks"),
    state("modal-fs", "is_open")
  ),
  function(n, is_open) {
    if (n > 0) {
      return(!is_open)
    }
    return(is_open)
  }
)

File: docs/components_page/components/modal/fullscreen.jl
using DashBootstrapComponents

modal = html_div([
    dbc_button("Open modal", id = "open-fs"),
    dbc_modal(
        [
            dbc_modalheader(dbc_modaltitle("Fullscreen modal")),
            dbc_modalbody("Wow this thing takes up a lot of space..."),
        ],
        id = "modal-fs",
        fullscreen = true,
    ),
])


callback!(
    app,
    Output("modal-fs", "is_open"),
    Input("open-fs", "n_clicks"),
    State("modal-fs", "is_open"),
) do n, is_open
    return n > 0 ? is_open == 0 : is_open
end;
